pkg/controller/core/workload_controller.go

```diff
@@ -248,6 +249,8 @@ func (r *WorkloadReconciler) reconcileCheckBasedEviction(ctx context.Context, wl
 	log := ctrl.LoggerFrom(ctx)
 	log.V(3).Info("Workload is evicted due to admission checks")
 	workload.SetEvictedCondition(wl, kueue.WorkloadEvictedByAdmissionCheck, "At least one admission check is false")
+	cqName, _ := r.queues.ClusterQueueFromLocalQueue(wl.Spec.QueueName)
+	metrics.ReportEvictedWorkloads(cqName, kueue.WorkloadEvictedByAdmissionCheck)
```

Comment:
Always report metrics after the Apply or Update call has succeeded. Otherwise, if there is failure with the API update, we will be counting double.

Reply:
this is still pending.

Reply:
All metrics is report after the patch is succeed now.